libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/utils/ParsedGradleVersion.kt

```diff
@@ -56,4 +57,12 @@ internal data class ParsedGradleVersion(val major: Int, val minor: Int) : Compar
 
 fun isGradleVersionAtLeast(major: Int, minor: Int) =
     ParsedGradleVersion.parse(GradleVersion.current().version)
-        ?.let { it >= ParsedGradleVersion(major, minor) } ?: false
\ No newline at end of file
+        ?.let { it >= ParsedGradleVersion(major, minor) } ?: false
+
+fun isConfigurationCacheAvailable(gradle: Gradle) =
+    try {
+        val startParameters = gradle.startParameter
+        startParameters.javaClass.getMethod("isConfigurationCache").invoke(startParameters) as? Boolean
```

Comment:
❗  that check is using reflection on internal API and exceptions for control flow and doesn't cache the result. This could have a bad effect on performance.